aspnet/publishing/linuxproduction.rst

```diff
@@ -0,0 +1,299 @@
+-------------
+
+1. Acess to an Ubuntu 14.04 Server with a standard user account with
+   sudo privilege.
+2. Public access to port 80 on your server. You may want to use IPTables
+   to restrict access to your server.
+3. An exisiting ASP.NET 5 application.
+
+Before getting started you require LibUV and DNVM. Follow the instructions on the :doc:`/getting-started/installing-on-linux` page.
+
+
+Copy over your app
+------------------
+
+Run ``dnu publish`` from your dev environment to package your
+application into a self-contained directory that can be launched.
+
+Before we proceed, go ahead and copy over your ASP.NET 5 application to
+your server using whatever tool (SCP, FTP, etc) integrates into your
+workflow. Try and run the app and navigate to http://yourserveraddress:port
+in your browser to see if the application runs fine on Linux. I
+recommend you have a working app before proceeding.
+
+If you do not have an application, I recommend using the `Yeoman
+generators <https://github.com/omnisharp/generator-aspnet>`__ to quickly
+scaffold a skeleton app.
+
+Configure a reverse proxy server
+--------------------------------
+
+A reverse proxy is a common setup for serving dynamic web applications.
+The reverse proxy terminates the HTTP request and forwards it to the
+ASP.NET application.
+
+Why use a reverse-proxy server?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Kestrel is great for serving dynamic content from ASP.NET, however the
+web serving parts aren’t as feature rich full-featured servers like IIS,
+Apache or Nginx. A reverse proxy-server can allow you to offload work
+like serving static content, caching requests, compressing requests, and
+SSL termination from the application server. The reverse proxy server
+may reside on a dedicated machine or may be deployed alongside an
+application server.
+
+For the purposes of this guide, we are going to use a single instance of
+Nginx, however based on your requirements you may chose a different
+setup.
+
+Install Nginx
+~~~~~~~~~~~~~
+
+.. code:: shell
+
+    sudo apt-get install nginx
+
+.. note::
+
+    If you plan to install optional Nginx modules you may be required to
+    build Nginx from source.
+
+We are going to ``apt-get`` to install Nginx. The installer also creates
+a System V init script that runs Nginx as daemon on system startup.
+Since we just installed Nginx for the first time, we can explicitly
+start it by running
+
+.. code:: shell
+
+    sudo service nginx start
+
+At this point you should be able to navigate to your browser and see the
+default landing page for Nginx.
+
+Configure Nginx
+~~~~~~~~~~~~~~~
+
+We will now configure Nginx as a reverse proxy to forward requests to
+our ASP.NET application
+
+We will be modifying the ``/etc/nginx/sites-available/default``, so open
+it up in your favorite text editor and replace the contents with the
+following.
+
+.. code::
+
+    server {
+        listen 80;
+
+        location / {
+            proxy_pass http://127.0.0.1:5000;
```

Comment:
Production config should probably use a UNIX socket rather than TCP, the performance of a UNIX socket is slightly better.